ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobMetadata.java

```diff
@@ -38,6 +41,19 @@
   private int containerId;
   private long expirationTime;
   private long deletionTime;
+  private String kmsContext;
+  private EncryptionType encryptionType;
+  private VcrEncryptionFormat vcrEncryptionFormat;
+
```

Comment:
Looks like we also need to store `CryptoService(Factory)` and `KMS(Factory)` in metadata. 

Because they are all varying between ambry and ambryLI. 

To encrypt a blob, we need `cryptoService`, `kmsService`, `kmsContext`, and `encryptionFormat`. Same requirements in decryption. So it's nice to store all of them in metadata. 

Storing class name is my preference, but also open to other suggestions.

Reply:
To keep integrity, `kmsScope` is also suggested to store in metadata.

Reply:
With this in mind, I think it's better to construct `cryptoAgent` in `CloudBlobStore` constructor instead of passing it in. 
Main reasons are:
1. This looks cleaner for CloudBackupManger` and `CloudBlobStore` constructor. 
2. CryptoAgent can be generated in fly by looking at metadata. 

It's OK to pass `vProps` to `CloudBackupManger` and then to `CloudBlobStore`.

Reply:
seems like adding all this information to cloudBlobMetadata may create a security risk (which is what encryption is attempting to protect against)

Reply:
I don't want to crowd the metadata with all that stuff.  How does the router do it?  Can't these parameters go into the blob block header so whoever consumes it later knows how to decrypt it?

Reply:
Yes. They can. @lightningrob